lib/subscription/subscription-store.ts

```diff
@@ -1,72 +1,47 @@
 /// <reference path='../../typings/tsd.d.ts' />
 
 import _ = require('lodash');
-import Subscription = require('./subscription');
+import Subscription = require('./subscription-buffer');
+import Map = require('../util/map');
+import interfaces = require('../interface');
 
 export = SubscriptionStore;
 
-class SubscriptionStore<T extends Subscription> {
+class SubscriptionStore extends Map<Subscription> {
```

Comment:
Yeah Let me examine all usage and see if it make sense to wrap the needed operations.